frontend/src/lib/k8s/resource.ts

```diff
@@ -0,0 +1,14 @@
+import { OpPatch } from 'json-patch';
+import { apiFactoryWithNamespace } from './apiProxy';
+
+export default class Resource {
+  static className = 'Resource';
```

Comment:
Is class `Resource` correct? We use the className for a number of operations, and "CustomResource" is a more appropriate name as Resource is too generic.

Reply:
This should be applied to the file name, class, and className BTW. So we can use `k8s.CustomResource` instead of `k8s.Resource`.